src/pools/thread/dynamic.ts

```diff
@@ -61,7 +61,10 @@ export class DynamicThreadPool<
     // All workers are busy, create a new worker
     const worker = this.createAndSetupWorker()
     this.registerWorkerMessageListener<Data>(worker, message => {
-      if (message.kill) {
+      const tasksInProgress = this.tasks.get(worker)
+      if (message.kill && !tasksInProgress) {
+        // Kill received from the worker, means that no new tasks are submitted to that worker for a while( > maxInactiveTime)
```

Comment:
Why do we have a comment here, but not in dynamic cluster?

Reply:
Now I added the same comment on the cluster